### PR TITLE
feat-811/sitemap-one-probe

### PR DESCRIPTION
**Pull request by lukman7788**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. add `--one-probe` options on sitemap (#811 )

## How did you implement / how did you fix it  
1.  run `/bin/run_dev --sitemap ./src/components/config/__tests__/sitemap.xml --one-probe`

## How to test  
1. test case was included (./src/components/config/__tests__/sitemap.xml) and expected result (./src/components/config/__tests__/expected.sitemap-oneprobe.yml)

image

### CODE REVIEW

File: docs/src/pages/guides/cli-options.md [hunk 1]
```diff
@@ -211,6 +211,12 @@ Monika supports reading sitemap.xml as configuration input. Use the `--sitemap`
 monika --sitemap sitemap.xml
 ```
 
+or you can also add `--one-probe` to merge all requests into single probe
```
Comment (nicnocquee) [inline docs/src/pages/guides/cli-options.md:214]: add `--one-probe` flag

File: src/index.ts [hunk 2]
```diff
@@ -154,6 +154,11 @@ class Monika extends Command {
       exclusive: ['har', 'insomnia', 'postman'],
     }),
 
+    'one-probe': Flags.boolean({
+      description: 'One Probe',
+      default: false,
```
Comment (nicnocquee) [inline src/index.ts:159]: please add `dependsOn: ['sitemap']` property

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/components/config/__tests__/expected.sitemap-oneprobe.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      - url: https://en.wiktionary.org/wiki/hyperjump
+        method: GET
+        timeout: 10000
+      - url: https://monika.hyperjump.tech/index
+        method: GET
+        timeout: 10000
+    interval: 900
+    incidentThreshold: 5
+    recoveryThreshold: 5
+    alerts: []
```
Comment (nicnocquee) [inline src/components/config/__tests__/expected.sitemap-oneprobe.yml:23]: fill the alerts with the default one: http not 200

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: docs/src/pages/guides/cli-options.md (hunk 1 above)
Comment (lukman7788) [reply docs/src/pages/guides/cli-options.md]: updated

File: src/components/config/__tests__/expected.sitemap-oneprobe.yml (hunk 3 above)
Comment (lukman7788) [reply src/components/config/__tests__/expected.sitemap-oneprobe.yml]: updated

**Review by nicnocquee — APPROVED** [review]

**Review by dennypradipta — APPROVED** [review]

**Pull request merged**